ios/NebulaNetworkExtension/PacketTunnelProvider.swift

```diff
@@ -56,19 +55,11 @@ class PacketTunnelProvider: NEPacketTunnelProvider {
         } catch {
             return completionHandler(error)
         }
-
-        let fileDescriptor = (self.packetFlow.value(forKeyPath: "socket.fileDescriptor") as? Int32) ?? -1
-        if fileDescriptor < 0 {
-            return completionHandler("Starting tunnel failed: Could not determine file descriptor")
-        }
-
-        var ifnameSize = socklen_t(IFNAMSIZ)
-        let ifnamePtr = UnsafeMutablePointer<CChar>.allocate(capacity: Int(ifnameSize))
-        ifnamePtr.initialize(repeating: 0, count: Int(ifnameSize))
-        if getsockopt(fileDescriptor, 2 /* SYSPROTO_CONTROL */, 2 /* UTUN_OPT_IFNAME */, ifnamePtr, &ifnameSize) == 0 {
-            self.ifname = String(cString: ifnamePtr)
+        
+        let fileDescriptor = tunnelFileDescriptor
+        if fileDescriptor == nil {
+            return completionHandler("Unable to locate the tun file descriptor")
```

Comment:
I see you added a bang down below. That bang implies that we could unwrap a nil and throw. That's not possible because of the nil check up here, but it's not readily obvious when reading the lower part of this function body.

Did you consider an approach like this, which would alleviate the need for the bang later on?

```
        if tunnelFileDescriptor == nil {
            return completionHandler("Unable to locate the tun file descriptor")
        }

        let fileDescriptor = tunnelFileDescriptor!
```